### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Nick13Zen**
First three homeworks.

### CODE REVIEW

File: HomeWork_1/src/Main.java [hunk 1]
```diff
@@ -0,0 +1,20 @@
+/**
+	 * Main class which contained entry point.
```
Comment (AlexanderSoroka) [inline HomeWork_1/src/Main.java]: Indentation is wrong

File: HomeWork_1/src/Main.java [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+	 * Prints input parameters in format: "Argument № = parameter" in reverse
+	 * order. Each parameter print in new line.
+	 * 
+	 * @author Nikita Zenchik
+	 */
+public class Main {
+	/**
```
Comment (AlexanderSoroka) [inline HomeWork_1/src/Main.java]: Indentation is wrong

File: HomeWork_1/src/Main.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+	 * The method that gets arguments from console line and prints the in reverse.
+	 */
+     public static void main(String[] args) {
+          for(int i=args.length-1; i>=0; i--) {
+             System.out.println("Arg " + i + " = " + args[i]);
+          }
+    }
```
Comment (AlexanderSoroka) [inline HomeWork_1/src/Main.java]: Indentation is wrong

File: HomeWork_1/src/Main.java [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+}
+
```
Comment (AlexanderSoroka) [inline HomeWork_1/src/Main.java]: remove these useless blank lines

File: HomeWork_2/src/Main.java [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+ * 
+ */
+
+public class Main {
+	/**
+	 * The method that gets user name from console and prints Hello Name.
+	 * @params args array with parameter as User Name.
+	 */
+	public static void main(String[] args) {
+        if(args.length>0)
```
Comment (AlexanderSoroka) [inline HomeWork_2/src/Main.java]: missed {}  after if and else

File: HomeWork_3/src/Calculate.java [hunk 6]
```diff
@@ -0,0 +1,55 @@
+     /**
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Calculate.java]: Indentation is wrong

File: HomeWork_3/src/CheckResult.java [hunk 7]
```diff
@@ -0,0 +1,63 @@
+     /**
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Indentation is wrong

File: HomeWork_3/src/CheckResult.java [hunk 8]
```diff
@@ -0,0 +1,63 @@
...
+     * The class contains methods for checking data.
+     */
+public class CheckResult {
+	/**
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Indentation is wrong

File: HomeWork_3/src/CheckResult.java [hunk 9]
```diff
@@ -0,0 +1,63 @@
...
+     */
+	public static String check(double number) {
+        if(Double.isNaN(number)) {
+            return Helper.NAN_NUMBER;
+        }
+        if(!Double.isInfinite(number)) {
+            return String.valueOf(number);
+        } else {
+            return Helper.BOUNDS_ERROR;
+
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: useless blank line

File: HomeWork_3/src/CheckResult.java [hunk 10]
```diff
@@ -0,0 +1,63 @@
...
+        }
+        if(!Double.isInfinite(number)) {
+            return String.valueOf(number);
+        } else {
+            return Helper.BOUNDS_ERROR;
+           
+        }
+    }
+	/**
+     * The method check if number is zero.
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java:21]: Indentation is wrong

File: HomeWork_3/src/CheckResult.java [hunk 11]
```diff
@@ -0,0 +1,63 @@
...
+     * 
+     * @param zero double value.
+     * @return True if zero==0.
+     * @return False if zero!=0.
+     */
+	public static boolean checkZero(double zero){
+		if(zero == 0) {	
+			System.out.println(Helper.DIVISION_BY_ZERO);
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Indentation is wrong

File: HomeWork_3/src/CheckResult.java [hunk 12]
```diff
@@ -0,0 +1,63 @@
...
+			return true;
+		}
+		else return false;
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Missed {} in else block

File: HomeWork_3/src/CheckResult.java [hunk 13]
```diff
@@ -0,0 +1,63 @@
...
+     * @param line string line.
+     * @return True if line=="-help".
+     * @return False if zero!="-help".
+     */
+	public static boolean checkHelp(String line){
+		if(line.equals("-help")) {
+			System.out.println(Helper.ABOUT);
+		   return true;
+		}
+		else return false;
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Missed {} in else block

File: HomeWork_3/src/CheckResult.java [hunk 14]
```diff
@@ -0,0 +1,63 @@
...
+	}
+	/**
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Indentation is wrong

File: HomeWork_3/src/CheckResult.java [hunk 15]
```diff
@@ -0,0 +1,63 @@
...
+     * The method check if line is empty.
+     * 
+     * @return True if line is empty.
+     * @return False if line is not empty.
+     */
+    public static boolean checkEmpty(String[] line){
+	  if(line.length==0) {
+
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: useless blank line

File: HomeWork_3/src/CheckResult.java [hunk 16]
```diff
@@ -0,0 +1,63 @@
...
+     * @return True if line is empty.
+     * @return False if line is not empty.
+     */
+    public static boolean checkEmpty(String[] line){
+	  if(line.length==0) {
+		  
+		System.out.println(Helper.EMPTY);
+	    return true;
+	  }
+	else return false;
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: missed {} in else block

File: HomeWork_3/src/Helper.java [hunk 17]
```diff
@@ -0,0 +1,31 @@
+
+public class Helper {
+	/**
+	 * The class contain some <code>String</code> constants for user info, help
+	 * and some errors. It contain command line commands for program.
+	 *
+	 * @author Nikuta Zenchik
+	 */
+	public static final String ABOUT = "The program perform " +
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Helper.java]: Indentation is wrong

File: HomeWork_3/src/Main.java [hunk 18]
```diff
@@ -0,0 +1,86 @@
...
+     * @param args array with command line launch parameters.
+     */
+    public static void main(String[] args) {
+        //try to read
+          try {
+        	  if(CheckResult.checkEmpty(args)){
+        		  System.exit(0);
+        	  }
+        	  else{
+                  String stringArg1 = args[0];
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Main.java]: Indentation is wrong

File: HomeWork_3/src/Main.java [hunk 19]
```diff
@@ -0,0 +1,86 @@
...
+                  if(CheckResult.checkHelp(args[0])){
+                	  System.exit(0);
+                  }
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Main.java:35]: } else {

File: HomeWork_3/src/Main.java [hunk 20]
```diff
@@ -0,0 +1,86 @@
...
+                  else{
+                      String stringArg2 = args[1];
+                      double arg1 = parsData(stringArg1);
+                      double arg2 = parsData(stringArg2);
+                      if(Double.isInfinite(arg1) &&
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Main.java]: Is it case when first is infinite and second is just number acceptable?

File: HomeWork_3/src/Main.java [hunk 21]
```diff
@@ -0,0 +1,86 @@
...
+                        Double.isInfinite(arg2)) {
+                	    System.out.println(Helper.BOUNDS_ERROR_INPUT);
+                        System.exit(0);
+                      } else {
+                	    printResult(arg1, arg2);
+                        }
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Main.java]: Indentation is wrong

File: HomeWork_3/src/Main.java [hunk 22]
```diff
@@ -0,0 +1,86 @@
...
+        System.out.println("Sum" + " = " + CheckResult.check(result));
+
+        result = Calculate.subtraction(arg1, arg2);
+        System.out.println("Subtraction" + " = " + CheckResult.check(result));
+
+        result = Calculate.multiplication(arg1, arg2);
+        System.out.println("Multiplication" + " = " + CheckResult.check(result));
+
+        if(CheckResult.checkZero(arg2))
+        	System.exit(0);
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Main.java]: missed {}

File: HomeWork_3/src/CheckResult.java [hunk 23]
```diff
@@ -0,0 +1,63 @@
...
+    }
+	/**
+     * The method check if number is zero.
+     * 
+     * @param zero double value.
+     * @return True if zero==0.
+     * @return False if zero!=0.
+     */
+	public static boolean checkZero(double zero){
+		if(zero == 0) {
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Invalid double comparison

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: HomeWork_2/src/Main.java [hunk 24]
```diff
@@ -15,10 +14,12 @@
 	 * @params args array with parameter as User Name.
 	 */
 	public static void main(String[] args) {
-        if(args.length>0)
+        if(args.length>0){
             System.out.println("Hello, " + args[0] + "!");
-        else
+        }
```
Comment (AlexanderSoroka) [inline HomeWork_2/src/Main.java]: here and everywhere must be } else [

File: HomeWork_3/src/CheckResult.java [hunk 25]
```diff
@@ -1,63 +1,66 @@
...
-	public static boolean checkZero(double zero){
-		if(zero == 0) {	
-			System.out.println(Helper.DIVISION_BY_ZERO);
-			return true;
-		}
-		else return false;
-	}
-	/**
+    public static boolean checkZero(double zero){
+         if(zero == 0) {
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Invalid double comparison

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: HomeWork_3/src/CheckResult.java [hunk 26]
```diff
@@ -25,29 +25,27 @@ public static String check(double number) {
      * @return False if zero!=0.
      */
     public static boolean checkZero(double zero){
-         if(zero == 0) {	
-           System.out.println(Helper.DIVISION_BY_ZERO);
+         if(!(Double.isInfinite(zero))) {
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: This check looks error-prone - zero is always finite I could not imagine what are you doing here

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: HomeWork_3/src/CheckResult.java (hunk 26 above)
Comment (Nick13Zen) [reply HomeWork_3/src/CheckResult.java]: Saw that after push. Fixed now

File: HomeWork_3/src/CheckResult.java [hunk 27]
```diff
@@ -0,0 +1,64 @@
...
+     * @param number double value.
+     */
+    public static String check(double number) {
+        if(Double.isNaN(number)) {
+           return Helper.NAN_NUMBER;
+        }
+        if(!Double.isInfinite(number)) {
+           return String.valueOf(number);
+        } else {
+               return Helper.BOUNDS_ERROR;
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/CheckResult.java]: Indentation is wrong - check it everywhere

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

**Review by AlexanderSoroka — APPROVED** [review]

File: HomeWork_4/src/Helper.java [hunk 28]
```diff
@@ -0,0 +1,33 @@
+/**
+ * The class contain some <code>String</code> constants for user info, help
+ * and some errors. It contain command line commands for program.
+ *
+ * @author Nikuta Zenchik
+ */
+public class Helper {
+
```
Comment (AlexanderSoroka) [inline HomeWork_4/src/Helper.java:8]: Useless blank line

File: HomeWork_4/src/Helper.java [hunk 29]
```diff
@@ -0,0 +1,33 @@
+/**
+ * The class contain some <code>String</code> constants for user info, help
+ * and some errors. It contain command line commands for program.
+ *
+ * @author Nikuta Zenchik
+ */
+public class Helper {
```
Comment (AlexanderSoroka) [inline HomeWork_4/src/Helper.java:7]: Inappropriate class name - Helper implies that it does some work and helps, but it contains only consts. something like Messages would be more clear.

File: HomeWork_4/src/Helper.java [hunk 30]
```diff
@@ -0,0 +1,33 @@
...
+
+    public static final String ABOUT = "The program solves the equation like ax^2 + bx + c = 0.";
+
+    public static final String ERROR_NOT_A_NUMBER = "Error. Input is " +
+            "incorrect. Only numbers are allowed. Separator between integral " +
+            "and fractional parts is \",\".";
+    public static final String EMPTY = "Error. Line you entered is empty \"-help\".";
```
Comment (AlexanderSoroka) [inline HomeWork_4/src/Helper.java:14]: Missed blank line

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: HomeWork_3/src/Main.java [hunk 31]
```diff
@@ -0,0 +1,80 @@
...
+     * subtraction,
+     * multiplication and division between it.
+     * 
+     * @param args array with command line launch parameters.
+     */
+    public static void main(String[] args) {
+        try {
+            if(CheckResult.checkEmpty(args)){
+               System.exit(0);
+            } else {
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Main.java:22]: You don't need this else

File: HomeWork_3/src/Main.java [hunk 32]
```diff
@@ -0,0 +1,80 @@
...
+               String stringArg1 = args[0];
+               if(CheckResult.checkHelp(args[0])){
+                  System.exit(0);
+               } else {
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Main.java:26]: Don't need this else

File: HomeWork_3/src/Main.java [hunk 33]
```diff
@@ -0,0 +1,80 @@
...
+                  String stringArg2 = args[1];
+                  double arg1 = parsData(stringArg1);
+                  double arg2 = parsData(stringArg2);
+                  if(Double.isInfinite(arg1) || Double.isInfinite(arg2)) {
+                     System.out.println(Helper.BOUNDS_ERROR_INPUT);
+                     System.exit(0);
+                  } else {
```
Comment (AlexanderSoroka) [inline HomeWork_3/src/Main.java:33]: Don't need this else

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]

File: HomeWork_3/src/Main.java (hunk 31 above)
Comment (Nick13Zen) [reply HomeWork_3/src/Main.java:22]: Home Work_3? I'm working on 4 and 5.

**Comment by AlexanderSoroka** [comment]
It contains a lot of useless files. I'll close it, you should merge all useless files to master and leave only 4-5 source code in dev, after create new pull-request - it must contain only java code for 4 and 5

**Pull request closed without merging**